driver-core/src/main/com/mongodb/internal/connection/SocketStream.java

```diff
@@ -91,6 +109,32 @@ protected Socket initializeSocket() throws IOException {
         throw new MongoSocketException("Exception opening socket", getAddress());
     }
 
+    private SSLSocket initializeSslSocketOverSocksProxy(final SSLSocketFactory sslSocketFactory) throws IOException {
+        String serverHost = address.getHost();
+        int serverPort = address.getPort();
+
+        SocksSocket socksProxy = new SocksSocket(null, settings.getProxySettings());
+        configureSocket(socksProxy, settings);
+        InetSocketAddress inetSocketAddress = InetSocketAddress.createUnresolved(serverHost, serverPort);
```

Comment:
Let's leave a comment expressing that it is crucial for the `InetSocketAddress` created from `this.address` to be unresolved, because we need it to be resolved by the proxy, instead of being resolved by us.

To have a single place where such a comment may be left, the code creating `InetSocketAddress` from `this.address` may be extracted into a method.